File: backend/internal/models/lancamento.go
package models

import "time"

type Lancamento struct {
	ID             int            `json:"id"`
	LojaID         int            `json:"loja_id"`
	Tipo           TipoLancamento `json:"tipo"`
	DepartamentoID int            `json:"departamento_id"`
	Data           time.Time      `json:"data_lancamento"`
	Observacao     *string        `json:"observacao,omitempty"`

	Itens []LancamentoItem `json:"itens"`
}

File: frontend/src/types/Lancamento.ts
export interface Lancamento {
  id?: number;
  departamentoID: number;
  tipo: string;
  data: Date;
  observacao?: string;
  itens: LancamentoItem[];
}

export interface LancamentoItem {
  produtoMerceariaID: number;
  produtoDepartamentoID: number;

  quantidade: number;
  unidadeMercearia: string;
  unidadeDepartamento: string;

  fatorConversao: number;
  totalLancado: number;
}

export interface LancamentoItemPayload {
  produtoMerceariaID: number;
  produtoDepartamentoID: number;
  quantidade: number;
}

export type LancamentoPayload = Omit<Lancamento, "itens"> & {
  itens: LancamentoItemPayload[];
};
